// react-native-font-faces/dist/commonjs/samples/Roboto.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Roboto_All = exports.Roboto_BlackItalic = exports.Roboto_Black = exports.Roboto_BoldItalic = exports.Roboto_Bold = exports.Roboto_MediumItalic = exports.Roboto_Medium = exports.Roboto_Italic = exports.Roboto_Regular = exports.Roboto_LightItalic = exports.Roboto_Light = exports.Roboto_ThinItalic = exports.Roboto_Thin = void 0;

var _FontSource = require("../types/FontSource");

const Roboto_Thin = {
  fontFamily: 'Roboto',
  fontWeight: '100',
  fontStyle: 'normal',
  src: [(0, _FontSource.local)('Roboto-Thin'), (0, _FontSource.url)('https://ff.static.1001fonts.net/r/o/roboto.thin.ttf', 'truetype')]
};
exports.Roboto_Thin = Roboto_Thin;
const Roboto_ThinItalic = {
  fontFamily: 'Roboto',
  fontWeight: '100',
  fontStyle: 'italic',
  src: [(0, _FontSource.local)('Roboto-ThinItalic'), (0, _FontSource.url)('https://ff.static.1001fonts.net/r/o/roboto.thin-italic.ttf', 'truetype')]
};
exports.Roboto_ThinItalic = Roboto_ThinItalic;
const Roboto_Light = {
  fontFamily: 'Roboto',
  fontWeight: '300',
  fontStyle: 'normal',
  src: [(0, _FontSource.local)('Roboto-Light'), (0, _FontSource.url)('https://ff.static.1001fonts.net/r/o/roboto.light.ttf', 'truetype')]
};
exports.Roboto_Light = Roboto_Light;
const Roboto_LightItalic = {
  fontFamily: 'Roboto',
  fontWeight: '300',
  fontStyle: 'italic',
  src: [(0, _FontSource.local)('Roboto-LightItalic'), (0, _FontSource.url)('https://ff.static.1001fonts.net/r/o/roboto.light-italic.ttf', 'truetype')]
};
exports.Roboto_LightItalic = Roboto_LightItalic;
const Roboto_Regular = {
  fontFamily: 'Roboto',
  fontWeight: '400',
  fontStyle: 'normal',
  src: [(0, _FontSource.local)('Roboto-Regular'), (0, _FontSource.url)('https://ff.static.1001fonts.net/r/o/roboto.regular.ttf', 'truetype')]
};
exports.Roboto_Regular = Roboto_Regular;
const Roboto_Italic = {
  fontFamily: 'Roboto',
  fontWeight: '400',
  fontStyle: 'italic',
  src: [(0, _FontSource.local)('Roboto-Italic'), (0, _FontSource.url)('https://ff.static.1001fonts.net/r/o/roboto.italic.ttf', 'truetype')]
};
exports.Roboto_Italic = Roboto_Italic;
const Roboto_Medium = {
  fontFamily: 'Roboto',
  fontWeight: '500',
  fontStyle: 'normal',
  src: [(0, _FontSource.local)('Roboto-Medium'), (0, _FontSource.url)('https://ff.static.1001fonts.net/r/o/roboto.medium.ttf', 'truetype')]
};
exports.Roboto_Medium = Roboto_Medium;
const Roboto_MediumItalic = {
  fontFamily: 'Roboto',
  fontWeight: '500',
  fontStyle: 'italic',
  src: [(0, _FontSource.local)('Roboto-MediumItalic'), (0, _FontSource.url)('https://ff.static.1001fonts.net/r/o/roboto.medium-italic.ttf', 'truetype')]
};
exports.Roboto_MediumItalic = Roboto_MediumItalic;
const Roboto_Bold = {
  fontFamily: 'Roboto',
  fontWeight: '700',
  fontStyle: 'normal',
  src: [(0, _FontSource.local)('Roboto-Bold'), (0, _FontSource.url)('https://ff.static.1001fonts.net/r/o/roboto.bold.ttf', 'truetype')]
};
exports.Roboto_Bold = Roboto_Bold;
const Roboto_BoldItalic = {
  fontFamily: 'Roboto',
  fontWeight: '700',
  fontStyle: 'italic',
  src: [(0, _FontSource.local)('Roboto-BoldItalic'), (0, _FontSource.url)('https://ff.static.1001fonts.net/r/o/roboto.bold-italic.ttf', 'truetype')]
};
exports.Roboto_BoldItalic = Roboto_BoldItalic;
const Roboto_Black = {
  fontFamily: 'Roboto',
  fontWeight: '900',
  fontStyle: 'normal',
  src: [(0, _FontSource.local)('Roboto-Black'), (0, _FontSource.url)('https://ff.static.1001fonts.net/r/o/roboto.black.ttf', 'truetype')]
};
exports.Roboto_Black = Roboto_Black;
const Roboto_BlackItalic = {
  fontFamily: 'Roboto',
  fontWeight: '900',
  fontStyle: 'italic',
  src: [(0, _FontSource.local)('Roboto-BlackItalic'), (0, _FontSource.url)('https://ff.static.1001fonts.net/r/o/roboto.black-italic.ttf', 'truetype')]
};
exports.Roboto_BlackItalic = Roboto_BlackItalic;
const Roboto_All = [Roboto_Thin, Roboto_ThinItalic, Roboto_Light, Roboto_LightItalic, Roboto_Regular, Roboto_Italic, Roboto_Medium, Roboto_MediumItalic, Roboto_Bold, Roboto_BoldItalic, Roboto_Black, Roboto_BlackItalic];
exports.Roboto_All = Roboto_All;
